backendutil/transform.go

```diff
@@ -0,0 +1,81 @@
+package backendutil
+
+import (
+	"io"
+
+	"github.com/emersion/go-smtp"
+)
+
+// TransformBackend is a backend that transforms messages.
+type TransformBackend struct {
+	Backend smtp.Backend
+
+	TransformMail func(from string) (string, error)
```

Comment:
It would be nice to add a usage example for it, wdyt?

Reply:
I just pushed some tests for the transform backend based on the existing server tests.

Since I am not using the TransformBackend anymore, I was not able to think of a good example. For the tests I just made the transform functions encode everything using Base64.